src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseParticipationService.java

```diff
@@ -252,4 +256,32 @@ public void stashChangesInStudentRepositoryAfterDueDateHasPassed(ProgrammingExer
             log.warn("Cannot stash student repository for participation {} because the repository was not copied yet!", participation.getId());
         }
     }
+
+    /**
+     * Replaces all files except the .git folder of the target repository with the files from the source repository
+     *
+     * @param targetURL the repository where all files should be replaced
+     * @param sourceURL the repository that should be used as source for all files
+     */
+    public void resetRepository(VcsRepositoryUrl targetURL, VcsRepositoryUrl sourceURL) throws GitAPIException, IOException {
+        Repository targetRepo = gitService.getOrCheckoutRepository(targetURL, true);
+        Repository sourceRepo = gitService.getOrCheckoutRepository(sourceURL, true);
+
+        // Replace everything but the .git folder
```

Comment:
```suggestion
        // Replace everything but the files corresponding to git (such as the .git folder or the .gitignore file)
```

Maybe you can change it to something like that